backend/internal/api/routes_services.go

```diff
@@ -25,13 +26,13 @@ func SetupRoutesWithServices(router *gin.Engine, nodeService *service.NodeServic
 		// Graph endpoints
 		api.GET("/graph", h.getGraph)
 		api.GET("/graph/viewport", h.getViewportNodes)
-		
+
 		// Node endpoints
 		api.GET("/nodes/:id", h.getNode)
 		api.GET("/nodes/:id/content", h.getNodeContent)
 		api.PUT("/nodes/:id/position", h.updateNodePosition)
 		api.GET("/nodes/search", h.searchNodes)
-		
+
 		// Vault management endpoints (for future use)
 		api.POST("/vault/parse", h.parseVault)
```

Comment:
TODO(CL): The route naming is inconsistent with RESTful conventions:
1. Current: POST /vault/parse → Should be: POST /vault/parses (create a new parse job)
2. Current: GET /vault/status → Should be: GET /vault/parses/latest or GET /vault/parses/:id

This would allow for:
- GET /vault/parses - List all parse history
- GET /vault/parses/:id - Get specific parse status
- DELETE /vault/parses/:id - Cancel a running parse